cookbooks/aws-parallelcluster-environment/spec/unit/recipes/cfn_bootstrap_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'spec_helper'
+
+describe 'aws-parallelcluster-environment::cfn_bootstrap' do
+  for_all_oses do |platform, version|
+    context "on #{platform}#{version}" do
+      cached(:cfnbootstrap_version) { '2.0-24' }
+      cached(:cfnbootstrap_package) { "aws-cfn-bootstrap-py3-#{cfnbootstrap_version}.tar.gz" }
+      cached(:python_version) { '3.7.16' }
+      cached(:system_pyenv_root) { 'system_pyenv_root' }
+      cached(:virtualenv_path) { "system_pyenv_root/versions/#{python_version}/envs/cfn_bootstrap_virtualenv" }
+
+      context "when cfn_bootstrap virtualenv not installed yet" do
+        cached(:chef_run) do
+          runner = ChefSpec::Runner.new(platform: platform, version: version) do |node|
+            node.override['cluster']['system_pyenv_root'] = system_pyenv_root
+            node.override['cluster']['region'] = 'non_china'
+          end
+          runner.converge(described_recipe)
+        end
+        cached(:node) { chef_run.node }
+
+        it 'installs pyenv for specific python version' do
+          is_expected.to run_install_pyenv_new('pyenv for cfn_bootstrap').with_python_version(python_version)
+        end
+
+        it 'activates cookbook vistualenv' do
+          is_expected.to run_activate_virtual_env('cfn_bootstrap_virtualenv').with(
+            pyenv_path: virtualenv_path,
+            python_version: python_version
+          )
+        end
+
+        it 'sets virtualenv path' do
```

Comment:
question, non-blocking, this seems to happen also if the virtual env is already installed. Same with the following steps.